x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticQueryBuilder.java

```diff
@@ -62,6 +68,35 @@ public class SemanticQueryBuilder extends AbstractQueryBuilder<SemanticQueryBuil
         args -> new SemanticQueryBuilder((String) args[0], (String) args[1], (Boolean) args[2])
     );
 
+    private static class LegacyInferenceResultsMap extends AbstractMap<String, InferenceResults> {
+        private static final String PLACEHOLDER_INFERENCE_ID = ".placeholder";
```

Comment:
Since you only ever call `getInferenceResults()` there's no need to define a placeholder key or have this be a functional map. Consider removing the defined placeholder key and have the `@Override` methods throw an `UnsupportedOperationException`.